scripts/zulip_emoji_merge_delegate.py

```diff
@@ -20,60 +20,61 @@
     site=ZULIP_SITE
 )
 
-# Fetch the last 200 messages
-response = client.get_messages({
-    "anchor": "newest",
-    "num_before": 200,
-    "num_after": 0,
-    "narrow": [{"operator": "channel", "operand": "PR reviews"}],
-})
-
-messages = response['messages']
-pr_pattern = re.compile(r'https://github\.com/leanprover-community/mathlib4/pull/(\d+)')
-
-for message in messages:
-    content = message['content']
-    match = pr_pattern.search(content)
-    if match:
-        pr_number = match.group(1)
-        # Check for emoji reactions
-        reactions = message['reactions']
-        has_peace_sign = any(reaction['emoji_name'] == 'peace_sign' for reaction in reactions)
-        has_bors = any(reaction['emoji_name'] == 'bors' for reaction in reactions)
-        has_merge = any(reaction['emoji_name'] == 'merge' for reaction in reactions)
-
-        pr_url = f"https://api.github.com/repos/leanprover-community/mathlib4/pulls/{pr_number}"
-        pr_response = requests.get(pr_url, headers={"Authorization": GITHUB_TOKEN})
-        pr_data = pr_response.json()
-        labels = [label['name'] for label in pr_data['labels']]
-
-        if 'delegated' in labels:
-            client.add_reaction({
-                "message_id": message['id'],
-                "emoji_name": "peace_sign"
-            })
-        elif 'ready-to-merge' in labels:
-            if has_peace_sign:
-                client.remove_reaction({
+for channel in ["PR reviews", "mathlib reviewers"]:
+    # Fetch the last 1000 messages
```

Comment:
I don't know the answer: does this fetch the latest 1000 messages overall on zulip, and then filter to the relevant channel --- or does this fetch the latest 1000 messages from that channel? (Because in the latter case, 1000 seems way overblown - this is run supposedly every hour.)

Reply:
I'm reasonably certain (because I just played with the API on a small test where I only got 2 messages) that it first narrows, then finds the anchor, then selects the messages. So, this gets the latest 1000 messages *within* the channel. So I agree with Michael, 1000 is too many. Even 200 seems high, but not absolutely impossible.

Reply:
If you want to test it yourself:
```bash
curl -sSX GET -G https://leanprover.zulipchat.com/api/v1/messages \
     -u <your@email.com>:<your-api-key> \
     --data-urlencode anchor=newest \
     --data-urlencode num_before=2 \
     --data-urlencode num_after=0 \
     --data-urlencode 'narrow=[{"operand": "Type theory", "operator": "channel"}]'
```